lib/pkg_editor/src/zlib.c

```diff
@@ -86,7 +86,12 @@ static void *zlib_load_library() {
   void *const library = (void *)LoadLibraryA(WINDOWS_ZLIB_PATH);
   return library;
 #else
-  return dlopen("libz.so", RTLD_NOW);
+  void *dlopen_handle = (void *)dlopen("libz.so.1", RTLD_NOW);
+  if (dlopen_handle == NULL) {
+    return dlopen("libz.so", RTLD_NOW);
+  } else {
+    return dlopen_handle;
+  }
```

Comment:
How about inverting the condition to make it slightly easier to follow the control flow?
```c++
void *const dlopen_handle = dlopen(...);
if (dlopen_handle) {
  return dlopen_handle;
}
return dlopen(...);
```
The early-return pattern is especially useful for more than two cases to avoid deep nesting of conditionals.

Reply:
(Applying our recent C++ training :slightly_smiling_face:, in C++17 the variable can be scoped to the conditional
```c++
if (void *const dlopen_handle = dlopen(...); dlopen_handle) {
  return dlopen_handle;
}
return dlopen(...);
```
but the runtime is currently built with C++11 where that construct is not available.)